view/play.php

```diff
@@ -4,24 +4,27 @@
 use Printer\CombiPrinter;
 use Printer\ResultPrinter;
 
-if (isset($solution, $compareResults, $propositions)) :
+if (isset($solution, $compareResults, $propositions, $colors)) :
 ?>
 
 <section id="section_play">
 
     <h1>Master Mind</h1>
 
+    <p>Ci-dessous les couleurs possibles pour cette partie :</p>
+
     <?php $combiPrinter = new CombiPrinter(); ?>
     <?php $resultPrinter = new ResultPrinter(); ?>
 
-    <?= $combiPrinter->printCombination($solution); ?>
+    <?= $combiPrinter->print($colors); ?>
+    <?= $combiPrinter->print($solution); ?>
 
     <?php for ($i = 0; $i < count($compareResults); $i++) : ?>
```

Comment:
Essaye de refactorer ce `for` avec un `foreach`

Reply:
J'ai déjà essayé pleins de fois hier avant que tu me le dises mais je n'ai jamais réussi.
Soit je fais le foreach sur $compareResults et $propositions pose problème soit l'inverse.

Reply:
Je pense que tu peux boucler comme ça 
```php
foreach (compareResult as $i => $compareResult) {
    echo $resultPrinter->printBlack($compareResult);
    echo $combiPrinter->print($propositions[$i]);
    echo $resultPrinter->printWhite($compareResult);
```
ou bien la même chose sur `$propositions` à toi de voir ;), mais j'ai vérifié tu as un gain de perf important sur le foreach donc c'est à utiliser.

Reply:
ça c'est ok. Je n'avais jamais vu cette façon de faire.
Pour la dernière fonction de CombiPrinter, de toute façon elle ne sert plus donc je l'ai supprimé.